### PR TITLE
Adding `step()` method for contexts in JS

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

We support a `context.step()` method in the legacy hEngine used within hCore/hCloud today, which provides the current step of the simulation. This PR implements the same in our new open hEngine.

## 🔗 Related links

- [Asana task description](https://app.asana.com/0/1199550852792314/1201525968072827/f)

## 🛡 Tests

- ✅ Manual Tests

## ❓ How to test this?

1. Download the step simulation from https://core.hash.ai/@alfie/steps/stable
2. Run a single_run with a few steps and check that the current_step field is correctly incremented in the outputted json_state.json

### CODE REVIEW

File: packages/engine/src/simulation/comms/mod.rs [hunk 1]
```diff
@@ -119,16 +119,18 @@ impl Comms {
         Ok(())
     }
 
+    /// TODO: docstring
```
Comment (litvand) [inline packages/engine/src/simulation/comms/mod.rs:122]: Could be DOC for easier search afterwards

**Review by litvand — APPROVED** [review]
LGTM

File: packages/engine/src/simulation/comms/mod.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/src/simulation/comms/mod.rs:122]: We should keep at least "TODO". "TODO DOC:" should work but I think "TODO: docstring" is fine, too.

Comment (Alfred-Mountfield) [reply packages/engine/src/simulation/comms/mod.rs:122]:
```suggestion
    /// TODO: DOC
```

**Pull request merged**